low-level/framework/utils/sysfs_interface.py

```diff
@@ -138,7 +140,144 @@ def fetch_nw_cable_status(self, nw_interface_path, interface):
                 carrier_indicator = cFile.read().strip()
             if carrier_indicator not in self.nw_phy_link_state.keys():
                 carrier_indicator = 'unknown'
-        except OSError as os_err:
-            return os_err.errno
+        except Exception:
```

Comment:
We need to catch specific exception like existing code and any generic exception handling happening in sspl_ll_d for each sensor.
Now since this function used for health collection, fine update specific exception to have "carrier_indicator = 'unknown'", but why to remove that exception all together ?

Reply:
If the file is not available we don't want to through Exception and exit right, for ex on vm, if we are running this, it should not exit from this point. We can not even print some msg like "file is not detected due to virtual environment" etc since we are serving the api's

Reply:
trying to understand expected behaviour, in case of Expection Handling, added comment in new PR.

Reply:
Raising Exception should only happen at rpath validation.
Otherwise the thread will be closed and data collection with sub resources will be lost.